pkg/server/skydb/pq/device.go

```diff
@@ -24,16 +24,46 @@ import (
 )
 
 func (c *conn) GetDevice(id string, device *skydb.Device) error {
-	builder := psql.Select("type", "token", "user_id", "last_registered_at").
+	builder := psql.Select("type", "token", "user_id", "topic", "last_registered_at").
 		From(c.tableName("_device")).
-		Where("id = ?", id)
+		Where("id = ? AND topic IS NOT NULL", id)
+
+	nullableToken := sql.NullString{}
+	nullableUserID := sql.NullString{}
+	err := c.QueryRowWith(builder).Scan(
+		&device.Type,
+		&nullableToken,
+		&nullableUserID,
+		&device.Topic,
+		&device.LastRegisteredAt,
+	)
+
+	if err == sql.ErrNoRows {
+		return skydb.ErrDeviceNotFound
+	} else if err != nil {
+		return err
+	}
+
+	device.Token = nullableToken.String
+	device.UserInfoID = nullableUserID.String
+	device.LastRegisteredAt = device.LastRegisteredAt.In(time.UTC)
+	device.ID = id
+
+	return nil
+}
+
+func (c *conn) GetDeviceByIDAndTopic(id, topic string, device *skydb.Device) error {
```

Comment:
Okay. It seems that the `GetDeviceByIDAndTopic` does not make sense. Let me remove it. And I will make the checking on handler instead of pusher, since I think it is handler's responsibility to query the devices and ask pusher to send push notification.